core/src/main/kotlin/com/strumenta/kolasu/serialization/JsonDeserializer.kt

```diff
@@ -38,7 +38,11 @@ class JsonDeserializer {
             when {
                 Node::class.java.isAssignableFrom(rawClass) -> {
                     val className = json.asJsonObject[JSON_TYPE_KEY].asString
-                    val actualClass = Class.forName(className)
+                    val actualClass = try {
+                        Class.forName(className)
+                    } catch (ex: ClassNotFoundException) {
+                        Class.forName(rawClass.`package`.name + "." + className)
```

Comment:
Why?

Reply:
Also, does not exist a method that check if a class exist without throwing an exception?

Reply:
I was just about to ask your opinion on this fix. It was needed to pass this test [deserializeNodeFromStreamingWithShortNames](https://github.com/Strumenta/kolasu/blob/24807dacbf6516772cf004c9c94ed13014d98be2/core/src/test/kotlin/com/strumenta/kolasu/serialization/JsonDeserializerTest.kt#L87-L103). It seems that sometimes with simple names cannot find the right class.

I will check for that method.

Reply:
There seems to be no such method. There is `forName` overload that does not throw an exception, but is Java 9+.

![image](https://github.com/Strumenta/kolasu/assets/12112299/b7a52c69-a89b-4f17-b8ba-14e2af1c2528)